File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        n = len(nums)
        left = right = pos = 0
        while right<n:
            while right<n and nums[left]==nums[right]:
                right+=1
            if right-left<3:
                while left<right:
                    nums[pos] = nums[left]
                    pos+=1
                    left+=1
            else:
                for i in range(2): nums[pos+i] = nums[left+i]
                pos += 2
                left = right
        return pos

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int left=0, right = 0, pos = 0, n = nums.length;
        while(right<n){
            while(right<n && nums[left]==nums[right]) right++;
            if(right-left<3){
                while(left<right) nums[pos++] = nums[left++];
            }
            else{
                nums[pos] = nums[left];
                nums[pos+1] = nums[left+1];
                pos += 2;
                left = right;
            }
        }
        return pos;
    }
}